.circleci/config.yml

```diff
@@ -248,23 +254,26 @@ jobs:
       - apply-patches
       - run:
           name: Run unit tests
-          command: cargo test --package << parameters.crate >> --all-features --lib -- --nocapture
+          command: cargo test -p << parameters.crate >> --all-features --lib -- --nocapture
       - run:
           name: Run integration tests
-          # Only run integration tests if there are any
           command: |
             set +o pipefail
-            (cargo test --package << parameters.crate >> --all-features --test '*' -- --list 2>&1 | grep -q "no test target matches pattern") \
-              && echo "nothing to test" \
-              || cargo test --package << parameters.crate >> --all-features --test '*' -- --nocapture
+            path=$(echo "<< parameters.crate >>" | sed "s/shuttle-\(.*\)/\1/")
+            if [ -f "$path/integration.sh" ]; then
+              bash "$path/integration.sh"
+            else
+              echo "nothing to test here, adjust CI file or add an integration script"
+              exit 1
+            fi
       - save-cargo-cache
-  test-workspace-member-integration-machine:
+  test-workspace-member-with-integration-xl:
     parameters:
       crate:
         description: "Crate in workspace to test"
         type: string
-    # Using a machine image since tests will start a docker container
-    executor: machine-ubuntu
+    executor: docker-rust
+    resource_class: xlarge
```

Comment:
nit: we can create a docker-rust-xlarge executor.

Reply:
I prefer it this way. I think this way is less DRY actually.

Reply:
I found a better way.